lib/mock-module/mock-module.ts

```diff
@@ -25,17 +26,23 @@ function MockIt(module: Type<NgModule>): IModuleOptions {
   }
   const mockedModule: IModuleOptions = { declarations: [],
                                          exports: [],
+                                         imports: [],
                                          providers: [] };
   const { declarations = [], imports = [], providers = [] } = ngModuleResolver.resolve(module);
 
   mockedModule.exports = mockedModule.declarations = [...declarations.map(MockDeclaration)];
   mockedModule.providers = providers.map(mockProvider);
+  mockedModule.imports = [];
 
   imports.forEach((imPort: Type<NgModule>) => {
     const result = MockIt(imPort);
-    mockedModule.declarations.push(...result.declarations);
-    mockedModule.providers.push(...result.providers);
-    mockedModule.exports.push(...result.declarations);
+    if ((result as any) === imPort) {
+      mockedModule.imports.push(imPort);
```

Comment:
in case of CommonModule we have to pass it as import to mocked module due to CommonModule is type and doesn't respect IModuleOptions interface.